Purchases/SwiftSources/LocalReceiptParsing/Builders/AppleReceiptBuilder.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Created by Andrés Boedo on 7/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+struct AppleReceiptBuilder {
+    private let containerBuilder: ASN1ContainerBuilder
+    private let inAppPurchaseBuilder: InAppPurchaseBuilder
+    private let dateFormatter: ISO3601DateFormatter
+
+    init() {
+        self.containerBuilder = ASN1ContainerBuilder()
+        self.inAppPurchaseBuilder = InAppPurchaseBuilder()
+        self.dateFormatter = ISO3601DateFormatter.shared
+    }
+
+    func build(fromASN1Container container: ASN1Container) throws -> AppleReceipt {
+        var bundleId: String?
+        var applicationVersion: String?
+        var originalApplicationVersion: String?
+        var opaqueValue: Data?
+        var sha1Hash: Data?
+        var creationDate: Date?
+        var expirationDate: Date?
+        var inAppPurchases: [InAppPurchase] = []
+
+        guard let internalContainer = container.internalContainers.first else { fatalError() }
+        let receiptContainer = try containerBuilder.build(fromPayload: internalContainer.internalPayload)
+        for receiptAttribute in receiptContainer.internalContainers {
+            let typeContainer = receiptAttribute.internalContainers[0]
```

Comment:
In the InAppPurchaseBuilder, there's a check for the number of internalContainers which I think would be helpful to add here